python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,1573 @@
+
+        Returns
+        -------
+        list
+            List of metadata values or defaults.
+        """
+        if metadata.exists(key):
+            return list(metadata.getArray(key))
+        else:
+            return [default] * numStamps
+
+    def _getDefaultVisitInfoDict(self, visitId: int | None = None) -> dict:
+        """Get a dictionary with default visit info values.
+
+        Parameters
+        ----------
+        visitId : int, optional
+            Visit ID to use. If None, uses UNKNOWN_INT.
+
+        Returns
+        -------
+        dict
+            Dictionary containing default visit info values.
+        """
+        return {
+            "visit_id": visitId if visitId is not None else self.UNKNOWN_INT,
+            "focus_z": self.UNKNOWN_FLOAT * u.mm,
+            "boresight_ra": self.UNKNOWN_FLOAT * u.deg,
+            "boresight_dec": self.UNKNOWN_FLOAT * u.deg,
+            "boresight_rot_angle": self.UNKNOWN_FLOAT * u.deg,
+            "boresight_par_angle": self.UNKNOWN_FLOAT * u.deg,
+            "boresight_alt": self.UNKNOWN_FLOAT * u.deg,
+            "boresight_az": self.UNKNOWN_FLOAT * u.deg,
+            "mjd": self.UNKNOWN_FLOAT,
+            "donut_radius": self.UNKNOWN_FLOAT,
+        }
+
+    def _determineDefocalType(self, exposure: afwImage.Exposure) -> str:
+        """Determine defocal type (intra/extra) from detector name.
+
+        Parameters
+        ----------
+        exposure : lsst.afw.image.Exposure
+            The exposure to analyze.
+
+        Returns
+        -------
+        str
+            String indicating defocal type: "intra" or "extra".
+
+        Raises:
+            ValueError: If defocal type cannot be determined from detector
+                name.
+        """
+        detectorName = exposure.getDetector().getName()
+        self.log.debug("Detector name: '%s'", detectorName)
+
+        # Determine defocal type based on detector name pattern
+        # SW0 detectors are extra-focal, SW1 detectors are intra-focal
+        # This matches the pattern used in CutOutDonutsCwfsTask
+        if detectorName.endswith("_SW0"):
+            defocalType = "extra"
+            self.log.debug("Detector '%s' is extra-focal (SW0)", detectorName)
+        elif detectorName.endswith("_SW1"):
+            defocalType = "intra"
+            self.log.info("Detector '%s' is intra-focal (SW1)", detectorName)
+        else:
+            # Only corner wavefront sensors (SW0/SW1) are supported
+            raise ValueError(
+                f"Cannot determine defocal type for detector '{detectorName}'. "
+                "This task only supports corner wavefront sensors with detector names "
+                "ending with '_SW0' (extra-focal) or '_SW1' (intra-focal). "
+                "Full array mode is not currently supported."
+            )
+
+        return defocalType
+
+    def _getExposureMetadata(self, exposure: afwImage.Exposure) -> dict[str, float | int]:
```

Comment:
If you need this info I think we should make a small change here https://github.com/lsst-ts/ts_wep/blob/982c2eab0d22739dd2bd7603ace8b1f224b2dffd/python/lsst/ts/wep/task/generateDonutCatalogUtils.py#L215-L272 to just give us a dictionary without the donut catalog part of it instead of recreating all the code here.